### PR TITLE
Email confirmation Functionality

### PR DESCRIPTION
**Pull request by irtazazulfiqar**

### CODE REVIEW

File: mydemoproject/demoapp1/apps.py [hunk 1]
```diff
@@ -5,3 +5,9 @@ class Demoapp1Config(AppConfig):
     default_auto_field = 'django.db.models.BigAutoField'
     name = 'demoapp1'
 
+    def ready(self):
+        import demoapp1.signals
+
+
+
+
```
Comment (m-shayan-ahmad) [inline mydemoproject/demoapp1/apps.py]: No extra line spaces.

File: mydemoproject/demoapp1/models/borrowed_book.py [hunk 2]
```diff
@@ -20,10 +21,16 @@ class BorrowedBook(BaseModel):
...
     return_dttm = models.DateTimeField(null=True, blank=True)
+    due_date = models.DateTimeField(null=True, blank=True)
 
     def __str__(self):
         return f"{self.user.username} borrowed {self.book.book_name}"
 
+    def save(self, *args, **kwargs):
+        if not self.due_date:
+            self.due_date = self.borrow_dttm + timedelta(weeks=1)
+        super().save(*args, **kwargs)
```
Comment (m-shayan-ahmad) [inline mydemoproject/demoapp1/models/borrowed_book.py]: There should be a single source of truth, and that is where we are adding the borrowed book. Also, this limit should be customizable via env. Default should be 1 week.

File: mydemoproject/demoapp1/signals.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        email_content = f"""
+        Dear {user.username},
+
+        Thank you for borrowing "{book.book_name}" by {book.author_name}.
+
+        You need to return it by {due_date.strftime('%Y-%m-%d')}.
+
+        Best regards,
+        Library Team
+        """
```
Comment (m-shayan-ahmad) [inline mydemoproject/demoapp1/signals.py:26]: Create a HTML template for this email.

File: mydemoproject/demoapp1/signals.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        # Send email using SendGrid
+        sg = sendgrid.SendGridAPIClient(api_key=settings.SENDGRID_API_KEY)
```
Comment (m-shayan-ahmad) [inline mydemoproject/demoapp1/signals.py]:
```suggestion
        sendgrid = sendgrid.SendGridAPIClient(api_key=settings.SENDGRID_API_KEY)
```

File: mydemoproject/demoapp1/signals.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            We need to disable SSL verification otherwise it will perform 
+            SSL verification
+        """
+
+        ssl._create_default_https_context = ssl._create_unverified_context
+
+        response = sg.send(mail)
+        print(response.status_code)
+        print(response.body)
+        print(response.headers)
```
Comment (m-shayan-ahmad) [inline mydemoproject/demoapp1/signals.py]: No print statements.

**Review by m-shayan-ahmad — CHANGES_REQUESTED** [review]

File: mydemoproject/demoapp1/signals.py (hunk 3 above)
Comment (irtazazulfiqar) [reply mydemoproject/demoapp1/signals.py:26]: catered in next PR

File: mydemoproject/demoapp1/models/borrowed_book.py [hunk 6]
```diff
@@ -46,6 +50,9 @@ def add_borrowed_book(cls, user, book):
             user=user,
             book=book,
             borrow_dttm=timezone.now(),
+            due_date=timezone.now() + timedelta(days=float(
+                os.getenv('EMAIL_DAYS')))
```
Comment (m-shayan-ahmad) [inline mydemoproject/demoapp1/models/borrowed_book.py:54]: Create .env.example file.

Comment (m-shayan-ahmad) [reply mydemoproject/demoapp1/models/borrowed_book.py:54]: Also, indent the parentheses correctly.

**Review by m-shayan-ahmad — APPROVED** [review]
There are some changes that must be taken care of in next PR.

**Pull request merged**